models/resource.go

```diff
@@ -0,0 +1,44 @@
+package models
+
+import (
+	"time"
+	"context"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"{{<service_name>}}/stores"
+	"github.com/labstack/echo"
+)
+
+type Resource struct {
+	ID        primitive.ObjectID `json:"id" bson:"_id"`
+	Name     string `json:"name" bson:"name"`
+	Code 	 string `json:"code" bson:"code"`
+	CreatedAt  time.Time     `json:"created_at" bson:"created_at"`
+	UpdatedAt  time.Time     `json:"updated_at" bson:"updated_at"`
+ 
+}
+
+// resource model constructor
+func NewResource() (*Resource, error) {
```

Comment:
Usually initializers doesn't return error, unless it receives a parameter and you rely on that to init the new instance, it's almost impossible to get an error here.

Reply:
Correct, typically a constructor only returns a pointer: `NewResource()*Resource`, however, you can return an error here if you need to validate anything. Typically, I would not validate in the constructor, and validate in the controller after the _object_ has been created.

Reply:
For sure, if you need to or think you will need to validate something then it's totally fine to return the error.